src/Raven.Server/Documents/Sharding/ShardedDocumentsStorage.cs

```diff
@@ -149,7 +149,11 @@ internal static ByteStringContext.Scope GenerateBucketAndEtagIndexKeyForDocument
     [StorageIndexEntryKeyGenerator]
     internal static ByteStringContext.Scope GenerateBucketAndEtagIndexKeyForTombstones(Transaction tx, ref TableValueReader tvr, out Slice slice)
     {
-        return GenerateBucketAndEtagIndexKey(tx, idIndex: (int)TombstoneTable.LowerId, etagIndex: (int)TombstoneTable.Etag, ref tvr, out slice);
+        var tombstoneType = *(Tombstone.TombstoneType*)tvr.Read((int)TombstoneTable.Type, out int _);
+        if (tombstoneType == Tombstone.TombstoneType.Revision)
```

Comment:
please check other types as well (but in case there is a problem and complex to fix open a dedicated issue)

Reply:
done - we need to do this for all tombstone types :)